assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -5411,7 +5411,7 @@ public void doView_submission(RunData data)
 				String userId = u == null ? "" : u.getId();
 				addAlert(state, rb.getFormattedMessage("cannotfin_submission_1", new String[]{assignmentReference, userId}));
 			}
-			if (submission != null)
+			if (submission != null && a.getContent().getTypeOfSubmission() != Assignment.NON_ELECTRONIC_ASSIGNMENT_SUBMISSION)
```

Comment:
a!=null (scrolling up to line 5402), and unlike a good bean, a.getContent() goes to the DB to retrieve the associated assignment_content object (which has to exist otherwise the assignment is broken)